### PR TITLE
[BUGFIX] Make EmbeddingCenterContextBatchify returned dtype robust to empty sentences

### PR DESCRIPTION
**Pull request by leezu**
## Description ##
np.asarray([]) defaults to float32 dtype. Thus an input of [[1,2,3,...], []]
would yield returnvalue of float32 dtype instead of expected int64 dtype.

### Essentials ###
- [X] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage
- [X] Code is well-documented

### Changes ###
- [X] Make EmbeddingCenterContextBatchify returned dtype robust to empty sentences

cc @dmlc/gluon-nlp-team

### CODE REVIEW

**Comment by mli** [comment]
Job PR-954/1 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-954/1/index.html

**Comment by sxjscience** [comment]
```python
import numpy as np
print(np.asarray([1,2,3]).dtype)
# int64
```

**Comment by sxjscience** [comment]
I see, it's to handle the np.asarray([]) case.

File: src/gluonnlp/data/batchify/embedding.py [hunk 1]
```diff
@@ -135,7 +135,8 @@ def __iter__(self):
         if isinstance(self._sentences[0][0], str):
             sentences = [np.asarray(s, dtype='O') for s in self._sentences]
         else:
-            sentences = [np.asarray(s) for s in self._sentences]
+            dtype = type(self._sentences[0][0])
```
Comment (sxjscience) [inline src/gluonnlp/data/batchify/embedding.py]: What if the first sentence is empty?

Comment (leezu) [reply src/gluonnlp/data/batchify/embedding.py]: Added itertools.chain to workaround empty first sentence

**Comment by mli** [comment]
Job PR-954/2 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-954/2/index.html

**Comment by mli** [comment]
Job PR-954/3 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-954/3/index.html

**Pull request merged**